### PR TITLE
feat(atomic): add support to close modal on escape

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/KIT-2218

### CODE REVIEW

**Review by ThibodeauJF — APPROVED** [review]

**Review by SimonMilord — APPROVED** [review]
Lgtm!

File: packages/atomic/src/components/common/atomic-modal/atomic-modal.tsx [hunk 1]
```diff
@@ -109,13 +110,28 @@ export class AtomicModal implements InitializableComponent<AnyBindings> {
     return classes;
   }
 
+  private handleCloseOnEscape() {
+    this.closeOnEscape = (e: KeyboardEvent) => {
+      if (e.key.toLowerCase() === 'escape') {
+        this.close();
+      }
+    };
+    document.body.addEventListener('keyup', this.closeOnEscape);
```
Comment (btaillon) [inline packages/atomic/src/components/common/atomic-modal/atomic-modal.tsx:119]: Can this be replaced with a `@Listen` decorator with the `document` target? https://stenciljs.com/docs/events#target

Comment (olamothe) [reply packages/atomic/src/components/common/atomic-modal/atomic-modal.tsx:119]: 🤔 Yeah probably !

File: packages/atomic/src/components/common/atomic-modal/atomic-modal.tsx [hunk 2]
```diff
@@ -89,6 +89,13 @@ export class AtomicModal implements InitializableComponent<AnyBindings> {
     }
   }
 
+  @Listen('keyup', {target: 'body'})
+  handleCloseOnEscape(e: KeyboardEvent) {
+    if (e.key.toLowerCase() === 'escape') {
```
Comment (louis-bompart) [inline packages/atomic/src/components/common/atomic-modal/atomic-modal.tsx:94]: 🤔 why lowerCase and not just "Escape" ?
https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values

Comment (olamothe) [reply packages/atomic/src/components/common/atomic-modal/atomic-modal.tsx:94]: Out of habit after all these years doing web dev, I always lower case compare anything that's coming from DOM API due to not knowing by heart where are all the landmines.

**Pull request merged**